src/views/Leads/GridItem/index.js

```diff
@@ -0,0 +1,291 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Link } from 'react-router-dom';
+import classNames from 'classnames';
+import Cloak from '#components/general/Cloak';
+import Button from '#rsca/Button';
+import DangerConfirmButton from '#rsca/ConfirmButton/DangerConfirmButton';
+import {
+    iconNames,
+    pathNames,
+} from '#constants';
+import { reverseRoute } from '#rsu/common';
+import {
+    leadTypeIconMap,
+    leadPaneTypeMap,
+    LEAD_PANE_TYPE,
+} from '#entities/lead';
+import _ts from '#ts';
+import { timeFrom } from '#utils/common';
+
+import leadThumbnail from '#resources/img/lead-thumbnail.png';
+import styles from './styles.scss';
+
+const propTypes = {
+    className: PropTypes.string,
+    lead: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+    style: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+    activeProject: PropTypes.number.isRequired,
+    itemIndex: PropTypes.number.isRequired,
+    onLeadClick: PropTypes.func.isRequired,
+    onSearchSimilarLead: PropTypes.func.isRequired,
+    onRemoveLead: PropTypes.func.isRequired,
+    onMarkProcessed: PropTypes.func.isRequired,
+    onMarkPending: PropTypes.func.isRequired,
+    minHeight: PropTypes.number, // eslint-disable-line react/no-unused-prop-types
+};
+
+const defaultProps = {
+    className: '',
+    style: {},
+    minHeight: 295,
+};
+
+const noop = () => {};
+
+export default class GridItem extends React.PureComponent {
+    static propTypes = propTypes;
+    static defaultProps = defaultProps;
+
+    // for masonry
+    static getItemHeight = (state, props) => {
+        const { width, minHeight } = state;
+        const {
+            thumbnailHeight = 400, thumbnailWidth = 250,
+        } = props;
+        // item is wrapped by masonry
+        const imgHeight = Math.max((width * thumbnailHeight) / thumbnailWidth, minHeight);
+        return imgHeight;
+    }
+
+    static shouldHideEntryAdd = ({ hasAnalysisFramework, entryPermissions }) => (
+        !hasAnalysisFramework || !(entryPermissions.create || entryPermissions.modify)
+    )
+
+    get links() {
+        const { activeProject, lead } = this.props;
+        return {
+            editLead: {
+                pathname: reverseRoute(
+                    pathNames.addLeads,
+                    {
+                        projectId: activeProject,
+                    },
+                ),
+                state: {
+                    serverId: lead.id,
+                    faramValues: lead,
+                },
+            },
+            editEntries: reverseRoute(
+                pathNames.editEntries,
+                {
+                    projectId: activeProject,
+                    leadId: lead.id,
+                },
+            ),
+        };
+    }
+
+    get mimeIcon() {
```

Comment:
Prefer not using getters, hard to tell if it will have a different reference while accessing that property.